library/cache-library/cache-service/src/main/java/uk/gov/gchq/gaffer/cache/CacheServiceLoader.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016-2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.cache;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.gchq.gaffer.cache.util.CacheSystemProperty;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+/**
+ * Bootstrapped when the REST service starts. Looks at a system property to determine the cache service to load.
+ * Then initialises it, after which any component may use {@code CacheServiceLoader.getService()} to get the service
+ * that can retrieve the appropriate cache.
+ */
+@WebListener(value = "Gaffer cache service loader")
```

Comment:
See other comment about removing this web listener and having it triggered by initialising a Gaffer Store.